src/registration/terms-of-service/TermsOfServicePage.test.ts

```diff
@@ -81,50 +80,41 @@ const setupMockAjax = async (
     termsOfService,
     enterpriseFeatures: [],
   };
-  const getTermsOfServiceText = jest.fn().mockResolvedValue('some text');
-  const getUserTermsOfServiceDetails = jest
-    .fn()
-    .mockResolvedValue(termsOfService satisfies SamUserTermsOfServiceDetails);
-  const acceptTermsOfService = jest.fn().mockResolvedValue(undefined);
-  const rejectTermsOfService = jest.fn().mockResolvedValue(undefined);
-  const getNihStatus = jest.fn();
-
-  type AjaxExports = typeof import('src/libs/ajax');
-  type AjaxContract = ReturnType<AjaxExports['Ajax']>;
-
-  asMockedFn(Ajax).mockImplementation(
-    () =>
-      ({
-        TermsOfService: {
-          acceptTermsOfService,
-          rejectTermsOfService,
-          getTermsOfServiceText,
-        },
-      } as DeepPartial<AjaxContract> as AjaxContract)
+  const getTermsOfServiceText: MockedFn<TermsOfServiceContract['getTermsOfServiceText']> = jest.fn();
+  getTermsOfServiceText.mockResolvedValue('some text');
+  const getUserTermsOfServiceDetails: MockedFn<TermsOfServiceContract['getUserTermsOfServiceDetails']> = jest.fn();
+  getUserTermsOfServiceDetails.mockResolvedValue(termsOfService satisfies SamUserTermsOfServiceDetails);
+  const acceptTermsOfService: MockedFn<TermsOfServiceContract['acceptTermsOfService']> = jest.fn();
+  acceptTermsOfService.mockResolvedValue(undefined);
+  const rejectTermsOfService: MockedFn<TermsOfServiceContract['rejectTermsOfService']> = jest.fn();
+  rejectTermsOfService.mockResolvedValue(undefined);
+  const getNihStatus: MockedFn<UserContract['getNihStatus']> = jest.fn();
+
+  asMockedFn(TermsOfService).mockReturnValue(
+    partial<TermsOfServiceContract>({
+      acceptTermsOfService,
+      rejectTermsOfService,
+      getTermsOfServiceText,
+      getUserTermsOfServiceDetails,
+    })
   );
 
-  asMockedFn(Metrics).mockReturnValue({
-    captureEvent: jest.fn(),
-  } as Partial<MetricsContract> as MetricsContract);
-
-  asMockedFn(User).mockReturnValue({
-    getSamUserCombinedState: jest.fn().mockResolvedValue(mockSamUserCombinedState),
-    profile: {
-      get: jest.fn().mockResolvedValue({ keyValuePairs: [] }),
-      update: jest.fn().mockResolvedValue({ keyValuePairs: [] }),
-      setPreferences: jest.fn().mockResolvedValue({}),
-      preferLegacyFirecloud: jest.fn().mockResolvedValue({}),
-    },
-    getNihStatus,
-  } as Partial<UserContract> as UserContract);
-
-  asMockedFn(Groups).mockReturnValue({
-    list: jest.fn(),
-  } as Partial<GroupsContract> as GroupsContract);
+  asMockedFn(Metrics).mockReturnValue(partial<MetricsContract>({ captureEvent: jest.fn() }));
+
+  asMockedFn(User).mockReturnValue(
+    partial<UserContract>({
+      getSamUserCombinedState: jest.fn().mockResolvedValue(mockSamUserCombinedState),
+      profile: {
+        get: jest.fn(async () => partial<TerraUserProfile>({})),
```

Comment:
this jest.fn inline form gives great type safety, whereas jest.fn().mockResolvedValue(...) has no type safety because of a cast to "any" out of jest.fn().